### PR TITLE
Remove `master` and `slave` from source code

### PR DESCRIPTION
**Pull request by PingXie**
External facing interfaces are not affected.

### CODE REVIEW

**Comment by hwware** [comment]
Before begin to take look this huge PR, can we setup several step for this. My suggestion as below:

1.  Set master as primary the alias name  (a new big change, but still backward compatible  )
2.  Remove master and slave from source code
3.  Update some json files, output message, and logs
4.  Update .md file and our valkey.io website

I suggest even this PR is approved, do not merge yet until step 1 is done.

**Comment by PingXie** [comment]
> Before begin to take look this huge PR, can we setup several step for this. My suggestion as below:
> 
> 1. Set master as primary the alias name  (a new big change, but still backward compatible  )
> 2. Remove master and slave from source code
> 3. Update some json files, output message, and logs
> 4. Update .md file and our valkey.io website
> 
> I suggest even this PR is approved, do not merge yet until step 1 is done.

@hwware I think 2 should be done first and this is what this PR is about. It should not impact any code behavior. thoughts?

**Comment by PingXie** [comment]
Also I don't think we can do much about 1 and 3 in the foreseeable future so really the order IMO is 

2,4,1,3

with 2 being the safest and should be done sooner rather than later (due to the sheer size of the change).

**Comment by hwware** [comment]
> > Before begin to take look this huge PR, can we setup several step for this. My suggestion as below:
> > 
> > 1. Set master as primary the alias name  (a new big change, but still backward compatible  )
> > 2. Remove master and slave from source code
> > 3. Update some json files, output message, and logs
> > 4. Update .md file and our valkey.io website
> > 
> > I suggest even this PR is approved, do not merge yet until step 1 is done.
> 
> @hwware I think 2 should be done first and this is what this PR is about. It should not impact any code behavior. thoughts?

My concern is developers have no idea where the term "primary" comes from (although most people can understand, primary is almost master)

**Comment by PingXie** [comment]
> My concern is developers have no idea where the term "primary" comes from (although most people can understand, primary is almost master)

We did document it in the source code already - https://github.com/valkey-io/valkey/blob/417660449f9c1fce998e1563f30f7bfd9cb5197f/src/valkeymodule.h#L10

But yeah I think we should scrub the documentation too

**Comment by hwware** [comment]
No problem, let us first go through this pr and then discuss other steps.

File: src/aof.c [hunk 1]
```diff
@@ -1073,7 +1073,8 @@ void flushAppendOnlyFile(int force) {
              * (because there's no reason, from the AOF POV, to call fsync) and then WAITAOF may wait on
```
Comment (hwware) [inline src/aof.c:1073]: I notice in this pr, you update comment as well, pls update the above line master_repl_offset to primary_repl_offset

File: src/aof.c [hunk 2]
```diff
@@ -1365,9 +1366,9 @@ struct client *createAOFClient(void) {
      */
```
Comment (hwware) [inline src/aof.c:1366]: pls update line 1358 "master"

File: src/aof.c [hunk 3]
```diff
@@ -2408,7 +2409,7 @@ int rewriteAppendOnlyFileBackground(void) {
```
Comment (hwware) [inline src/aof.c:2409]: line 2406

**Review by hwware — APPROVED** [review]

**Comment by hwware** [comment]
Let us merge and then document it.

**Pull request merged**